### PR TITLE
Add SimulationParameterChangeTracker for compound parameter tracking

### LINKED ISSUE #3430: Change tracking for simulation parameters
## Parent PRD

#3422

## What to build

Implement tracking of uncommitted compound-dependent simulation parameter changes. This enables the visual indicator and commit workflow.

**SimulationParameterChangeTracker class:**
- Maintains list of ObjectPath for changed simulation parameters
- `HasUncommittedChanges` property (true when list is non-empty)
- Methods: Track(path), Untrack(path), Clear(), GetChangedPaths()

**Integration with simulation:**
- Add `SimulationParameterChangeTracker` to Simulation
- Persisted with simulation (survives save/load)

**Command integration:**
- Commands that modify compound-dependent simulation parameters add paths to tracker
- Uses path matching logic from #3427 to identify compound-dependent parameters
- Reset parameter command removes path from tracker

**Path matching behavior:**
- Only track parameters that belong to compounds used as building blocks
- Do not track parameters for unnamed metabolites or other simulation-created compounds

## Acceptance criteria

- [ ] `SimulationParameterChangeTracker` class exists with Track/Untrack/Clear/GetChangedPaths methods
- [ ] `HasUncommittedChanges` property returns true when there are tracked changes
- [ ] Tracker is attached to Simulation
- [ ] Tracker persists with project save/load
- [ ] Modifying a compound-dependent simulation parameter adds it to tracker
- [ ] Resetting a parameter removes it from tracker
- [ ] Parameters for unnamed metabolites are not tracked
- [ ] Unit tests for tracker behavior

## Blocked by

- #3427

## User stories addressed

- User story 2: visual indicator for uncommitted changes (foundation)
- User story 19: reset parameter removes from uncommitted changes list
- User story 20: uncommitted changes persisted when saving project

### PR DESCRIPTION
**Pull request by msevestre**
## Summary
- Adds `SimulationParameterChangeTracker` class to track uncommitted compound-dependent simulation parameter changes
- Includes BDD unit tests covering: track, untrack, clear, dedup, and empty state scenarios

## References
- Fixes #3430
- Parent PRD: #3422
- Depends on: #3427

## Test plan
- [ ] CI tests pass for `SimulationParameterChangeTrackerSpecs`
- [ ] T2-T4 (Simulation integration, command integration) to follow in subsequent commits

## Release Notes

* **New Features**
  * Added parameter change tracking to simulations, enabling identification of modified parameters awaiting commit.
  * Parameter change data is preserved across simulation clones and serialization operations.

* **Tests**
  * Comprehensive test coverage for parameter change tracking functionality, including edge cases for undo/redo operations.

### CODE REVIEW

**Comment by msevestre** [comment]
@coderabbitai full review

**Comment by msevestre** [comment]
@Yuri05 @rwmcintosh Building upon the existing code. This is the code that is tracking changes in the simulation. I am not sure at this stage if "path" is enough or if we need to also store the compound name. I opted for not storing the component for now as it's a simple quick lookup.

File: src/PKSim.Core/Commands/EditParameterCommand.cs [hunk 1]
```diff
@@ -132,6 +182,7 @@ public override void UpdateInternalFrom(IBuildingBlockChangeCommand originalComm
          base.UpdateInternalFrom(originalCommand);
          var editChangeCommand = originalCommand.DowncastTo<EditParameterCommand>();
          _alteredOn = editChangeCommand._alteredOn;
+         _trackedByThisCommand = editChangeCommand._trackedByThisCommand;
```
Comment (msevestre) [inline src/PKSim.Core/Commands/EditParameterCommand.cs:185]: @rwmcintosh @Yuri05 This is the trick to be able to undo the command properly

**Comment by rwmcintosh** [comment]
> @Yuri05 @rwmcintosh Building upon the existing code. This is the code that is tracking changes in the simulation. I am not sure at this stage if "path" is enough or if we need to also store the compound name. I opted for not storing the component for now as it's a simple quick lookup.

You left a TODO for that in the code but if we don't need it do you think the TODO will be revisited again? Otherwise maybe remove the TODO and if we need it we'll find out then. Just to avoid a TODO comment that doesn't get cleaned up.

**Review by rwmcintosh — APPROVED** [review]

File: src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs [hunk 2]
```diff
@@ -122,6 +122,7 @@ public override async Task<SnapshotSimulation> MapToSnapshot(ModelSimulation sim
          snapshot.HasResults = simulation.HasResults;
          snapshot.AlteredBuildingBlocks = alteredBuildingBlocksIn(simulation);
          snapshot.OverwriteParameterSetSelections = await _overwriteParameterSetSelectionMapper.MapToSnapshots(simulation.OverwriteParameterSetSelections.Selections, project);
+         snapshot.ChangedParameterPaths = changedParameterPathsFrom(simulation);
```
Comment (Yuri05) [inline src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs]: Do we need to save these parameter paths separately in the snapshot?

This would be a subset of parameter paths for `snapshot.Parameters` (L117), correct? 
So when loading a simulation from snapshot, we could reanalyze each of them for the tracker (for the older snapshots created with ≤v12, this needs to be done anyway).

Advantage of the current solution: no need to reanalyze parameters changed in the simulation (the time gain during the simulation creation is probably not that high though). 
Disadvantages: duplication of the information in the snapshot; direct snapshot manipulation (e.g. via [osp.snapshots](https://github.com/esqLABS/osp.snapshots) etc. becomes tricky and error-prone).

Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs]: Hum I did not think about this. So you are suggesting to not saved this information at all and rather reconstruct? What would it be? we load all the changes parameters in the simulation, see if they are "Compound" matchng (e.g SImulation type and belong to compound) and add them ?

Comment (Yuri05) [reply src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs]: > Hum I did not think about this. So you are suggesting to not saved this information at all and rather reconstruct?

exactly.

> What would it be? we load all the changes parameters in the simulation, see if they are "Compound" matchng (e.g SImulation type and belong to compound) and add them ?

Yes, something like this.
